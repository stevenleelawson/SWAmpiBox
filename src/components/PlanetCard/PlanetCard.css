.planet-name,
.planet-terrain,
.planet-population,
.planet-climate,
.planet-residents {
  color: white;
}

.planet {
  border: 3px solid #fff;
  border-radius: 5px;
  height: 20rem;
  width: 15rem;
  margin: 5rem;
}

.active {
  background-color: magenta;
}
