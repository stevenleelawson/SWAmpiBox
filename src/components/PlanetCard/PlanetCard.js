import React from 'react';
import './PlanetCard.css';

const PlanetCard = ({name, terrain, population, climate, residents, toggleFavorites}) => {
  return (
    <div className='planet'>
      <h1 className='planet-name'>{name}</h1>
      <p className='planet-terrain'>terrain: {terrain}</p>
      <p className='planet-population'>population: {population}</p>
      <p className='planet-climate'>climate: {climate}</p>
      <p className='planet-residents'>residents: {residents}</p>
      <button onClick={() => toggleFavorites({name, terrain, population, climate, residents, category: 'planets'})}>Favorite</button>
    </div>
  )
}

export default PlanetCard;
