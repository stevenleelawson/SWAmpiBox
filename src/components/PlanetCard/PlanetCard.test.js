import React from 'react';
import { shallow } from 'enzyme';
import PlanetCard from './PlanetCard';

describe('PlanetCard', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<PlanetCard />)
    expect(wrapper).toMatchSnapshot();
  })
})
