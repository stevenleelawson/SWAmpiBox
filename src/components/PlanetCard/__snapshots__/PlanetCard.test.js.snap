// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlanetCard should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PlanetCard />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="planet-name"
/>,
        <p
          className="planet-terrain"
>
          terrain: 
</p>,
        <p
          className="planet-population"
>
          population: 
</p>,
        <p
          className="planet-climate"
>
          climate: 
</p>,
        <p
          className="planet-residents"
>
          residents: 
</p>,
        <button
          className="button"
          onClick={[Function]}
>
          Favorite
</button>,
      ],
      "className": "planet",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "planet-name",
        },
        "ref": null,
        "rendered": null,
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "terrain: ",
            undefined,
          ],
          "className": "planet-terrain",
        },
        "ref": null,
        "rendered": Array [
          "terrain: ",
          undefined,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "population: ",
            undefined,
          ],
          "className": "planet-population",
        },
        "ref": null,
        "rendered": Array [
          "population: ",
          undefined,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "climate: ",
            undefined,
          ],
          "className": "planet-climate",
        },
        "ref": null,
        "rendered": Array [
          "climate: ",
          undefined,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "residents: ",
            undefined,
          ],
          "className": "planet-residents",
        },
        "ref": null,
        "rendered": Array [
          "residents: ",
          undefined,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Favorite",
          "className": "button",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Favorite",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="planet-name"
/>,
          <p
            className="planet-terrain"
>
            terrain: 
</p>,
          <p
            className="planet-population"
>
            population: 
</p>,
          <p
            className="planet-climate"
>
            climate: 
</p>,
          <p
            className="planet-residents"
>
            residents: 
</p>,
          <button
            className="button"
            onClick={[Function]}
>
            Favorite
</button>,
        ],
        "className": "planet",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "planet-name",
          },
          "ref": null,
          "rendered": null,
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "terrain: ",
              undefined,
            ],
            "className": "planet-terrain",
          },
          "ref": null,
          "rendered": Array [
            "terrain: ",
            undefined,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "population: ",
              undefined,
            ],
            "className": "planet-population",
          },
          "ref": null,
          "rendered": Array [
            "population: ",
            undefined,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "climate: ",
              undefined,
            ],
            "className": "planet-climate",
          },
          "ref": null,
          "rendered": Array [
            "climate: ",
            undefined,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "residents: ",
              undefined,
            ],
            "className": "planet-residents",
          },
          "ref": null,
          "rendered": Array [
            "residents: ",
            undefined,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Favorite",
            "className": "button",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Favorite",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
